go/cmd/dolt/commands/sql.go

```
@@ -676,6 +653,15 @@ func runShell(ctx *sql.Context, se *sqlEngine, mrEnv env.MultiRepoEnv) error {
 			}
 		}
 
+		if err == nil {
+			verr := writeRoots(ctx, se, mrEnv, initialRoots)
+
+			if verr != nil {
```

You want to return this error from runShell, not just print it